core/src/main/java/org/apache/iceberg/actions/RewritePositionDeletesCommitManager.java

```diff
@@ -0,0 +1,129 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import java.util.Set;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.RewriteFiles;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.exceptions.CommitStateUnknownException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.apache.iceberg.util.Tasks;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Functionality used by RewritePositionDeleteFile Actions from different platforms to handle
+ * commits.
+ */
+public class RewritePositionDeletesCommitManager {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(RewritePositionDeletesCommitManager.class);
+
+  private final Table table;
+  private final long startingSnapshotId;
+
+  public RewritePositionDeletesCommitManager(Table table) {
+    this.table = table;
+    this.startingSnapshotId = table.currentSnapshot().snapshotId();
+  }
+
+  /**
+   * Perform a commit operation on the table adding and removing files as required for this set of
+   * file groups
+   *
+   * @param fileGroups fileSets to commit
+   */
+  public void commitFileGroups(Set<RewritePositionDeleteGroup> fileGroups) {
+    Set<DeleteFile> rewrittenDeleteFiles = Sets.newHashSet();
+    Set<DeleteFile> addedDeleteFiles = Sets.newHashSet();
+    for (RewritePositionDeleteGroup group : fileGroups) {
+      rewrittenDeleteFiles.addAll(group.rewrittenDeleteFiles());
+      addedDeleteFiles.addAll(group.addedDeleteFiles());
+    }
+
+    RewriteFiles rewrite = table.newRewrite().validateFromSnapshot(startingSnapshotId);
+    rewrite.rewriteDeleteFiles(rewrittenDeleteFiles, addedDeleteFiles);
+
+    rewrite.commit();
+  }
+
+  /**
+   * Clean up a specified file set by removing any files created for that operation, should not
+   * throw any exceptions
+   *
+   * @param fileGroup group of files which has already been rewritten
+   */
+  public void abortFileGroup(RewritePositionDeleteGroup fileGroup) {
+    Preconditions.checkState(
+        fileGroup.addedDeleteFiles() != null, "Cannot abort a fileGroup that was not rewritten");
+
+    Tasks.foreach(fileGroup.addedDeleteFiles())
```

Comment:
Shall we use bulk deletes if possible? Can we use `CatalogUtil$deleteFiles`?

Reply:
Exposed the method as public in CatalogUtil and now I use it